### PR TITLE
Union of two array

### PR DESCRIPTION
**Pull request by Mugdha-Hazra**
Added code for issue  #739 finding the Union Of Two Arrays, @tarun26091999 @anubhavitis @veenit1802 @ssaksham @malavikarajeshvikraman  sir can you please review my code.

### CODE REVIEW

**Comment by rish-singhal** [comment]
Please mention the issue number.

**Comment by Mugdha-Hazra** [comment]
issue no. is #739

File: Code/C++/Generate_all_valid_parentheses.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/*
+Problem statement : Given n pairs of parentheses, we have to write a function to generate all combinations of well-formed parentheses.
+Constraints:
```
Comment (rish-singhal) [inline Code/C++/Generate_all_valid_parentheses.cpp:3]: Can you please remove this file, as it is not the part of the ISSUE

File: Code/C++/finding_all_subsets.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/*
+FINDING ALL THE SUBSET OF THE ARRAY/VECTOR BY USING BACKTRACKING
```
Comment (rish-singhal) [inline Code/C++/finding_all_subsets.cpp:2]: Remove this file too

File: Code/C++/Union_of_two_unsorted_array.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/*
+Sample Input output set
+==========================
+
+Sample Input
+5
+6
+2 6 9 7 1
+1 2 3 4 5 6
+Sample Output
```
Comment (rish-singhal) [inline Code/C++/Union_of_two_unsorted_array.cpp:55]: Give time complexity and at least one more test case

File: Code/C++/Union_of_two_unsorted_array.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// CPP program to find the Union of two unsorted arrays.
+
+/* Procedure 
+==================
+First of all here we'll initialize an empty hash set 's'.
+then we'll iterate through the first array and put every element of the first array in the set s.
+Repeat the process for the second array.
+Finally we'll print the set s.
+Time Complexity: O(m+n)
```
Comment (rish-singhal) [inline Code/C++/Union_of_two_unsorted_array.cpp:9]: This is not O(m+n) as you are using set, for inserting something in set it takes O(log n) time complexity.

So, please update it accordingly and shift this to the bottom of the code.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by rish-singhal** [comment]
Also, update read me adding link to this file.

**Comment by Mugdha-Hazra** [comment]
Hello @rish-singhal  sir, I had tried to make all the changes that you had sited above but I'm a Lil confused about where to edit the readme can you please help me out as I'm a complete noob in this field, also can you please review the changes that I had made.

**Comment by rish-singhal** [comment]
This is the link to the [readme](https://github.com/Algo-Phantoms/Algo-Tree/tree/main/Array)

File: Code/C++/Union_of_two_unsorted_array.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        cin >> a2[i];
+    }
+    cout << "\nUnion of both the arrays:\n";
+    Union(a1, a2, n1, n2);
+    return 0;
+}
+
+/*
+Time Complexity: O(log n)
```
Comment (rish-singhal) [inline Code/C++/Union_of_two_unsorted_array.cpp:45]: Time complexity is O((n+m) log(n+m))

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Mugdha-Hazra** [comment]
@rish-singhal sir I had updated the readme.

**Comment by rish-singhal** [comment]
hey don't create another PR for this. Update readme in this PR only.

**Comment by Mugdha-Hazra** [comment]
@rish-singhal sir can you please guide me on how to update this pr for updating the readme?

**Comment by rish-singhal** [comment]
Update the readme in this branch of yours only.

**Comment by Mugdha-Hazra** [comment]
Hello @rish-singhal sir, I tried a lot to update the readme but I'm facing some errors can you plz help me to resolve these issues.

**Comment by Mugdha-Hazra** [comment]
Hello @rish-singhal sir, I finally updated the readme.

File: Code/C++/Union_of_two_unsorted_array.cpp [hunk 6]
```diff
@@ -0,0 +1,87 @@
+// CPP program to find the Union of two unsorted arrays.
+
+/* Procedure 
+==================
+First of all here we'll initialize an empty hash set 's'.
```
Comment (rish-singhal) [inline Code/C++/Union_of_two_unsorted_array.cpp:5]: Hey just remove the word hash, as set is not a hash based data structure. Then your code is ready to merge!

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Mugdha-Hazra** [comment]
I removed the hash word

**Comment by Mugdha-Hazra** [comment]
Thank you soo much @rish-singhal  sir for your guidance and support today I learned a lot about Git and GitHub and successfully able to make my first open-source contribution.

**Comment by rish-singhal** [comment]
Great, all the best for your open source journey :)

**Pull request merged**